### DISCUSSION TITLE
Component to render some code only client side

### DISCUSSION THREAD

**Discussion by fabiorodriguesroque** [inertiajs/inertia#1722] [Ideas]
Some components we just want to render by client side even if we are working with ssr. 

could be a good idea have a component to force render only in client side?

eg:
```
<inertia-client>
  <slot>
</inertia-client>
```

This is my first time starting a discussion here, so let me know if I did something wrong.

**Comment by evan-burrell** [comment]
I have opened #3218 implementing this for React, Vue and Svelte.

`<ClientOnly>` renders its children only after the component has mounted in the browser, with a `fallback` slot rendered during SSR and the initial hydration render:

```vue
<ClientOnly>
  <Chart />
  <template #fallback><p>Loading chart...</p></template>
</ClientOnly>
```

The existing workarounds each give up something. Disabling SSR per endpoint via `inertia.ssr.enabled` (as suggested in #1429) costs you server rendering for the whole page to accommodate one widget, and leaning on `Deferred` requires inventing a deferred prop plus an HTTP round-trip to render something that needs no server data. This keeps SSR for the page and carves out just the subtree.

One limitation worth stating up front, since the name invites the Nuxt comparison: Nuxt also tree-shakes the wrapped component out of the server build. This is runtime-only, so the children's module is still imported server-side, it just is not rendered. That covers the `window`/`document` access case in this thread, but it will not strip a server-incompatible dependency out of the SSR bundle.

Tests cover both the SSR and client-rendered paths across all three adapters. Feedback on the API shape very welcome — in particular whether `fallback` should also accept a tag/string form the way Nuxt's `fallbackTag` and `placeholder` do.
